src/main/java/org/broadinstitute/hellbender/tools/reference/CheckReferenceCompatibility.java

```diff
@@ -0,0 +1,284 @@
+    public GATKPath vcfPath;
+
+    /**
+     * Output file will be written here.
+     *
+     * Note: If no output file provided, table will print to standard output.
+     */
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "If specified, output reference sequence table in TSV format to this file. Otherwise print it to stdout.", optional = true)
+    private GATKPath output;
+
+    private SAMSequenceDictionary dictionaryToCompare;
+    private GATKPath dictionaryPath;
+    private String dictionaryName;
+    private Map<GATKPath, SAMSequenceDictionary> dictionaries = new LinkedHashMap<>();
+    private List<CompatibilityRecord> referenceCompatibilities = new ArrayList<>();
+
+    @Override
+    protected SequenceDictionaryValidationArgumentCollection getSequenceDictionaryValidationArgumentCollection() {
+        return new SequenceDictionaryValidationArgumentCollection.NoValidationCollection();
+    }
+
+    @Override
+    public void onTraversalStart() {
+        initializeSequenceDictionaryForInput();
+    }
+
+    private void initializeSequenceDictionaryForInput(){
+        // BAMs/CRAMs
+        if(hasReads() ^ vcfPath != null){
+            if (hasReads()) {
+                if (readArguments.getReadPathSpecifiers().size() > 1) {
+                    throw new UserException.BadInput("Tool analyzes one BAM at a time.");
+                }
+                dictionaryToCompare = getHeaderForReads().getSequenceDictionary();
+                dictionaryPath = readArguments.getReadPathSpecifiers().get(0);
+                dictionaryName = dictionaryPath.toPath().getFileName().toString();
+            }
+            // VCFs
+            else {
+                try(final FeatureDataSource<VariantContext> vcfReader = new FeatureDataSource<>(vcfPath.toString())){
+                    VCFHeader header = (VCFHeader) vcfReader.getHeader();
+                    dictionaryToCompare = header.getSequenceDictionary();
+                    dictionaryPath = vcfPath;
+                    dictionaryName = dictionaryPath.toPath().getFileName().toString();
+                }
+            }
+            dictionaries.put(dictionaryPath, dictionaryToCompare);
+            for(GATKPath path : references){
+                dictionaries.put(path, ReferenceDataSource.of(path.toPath()).getSequenceDictionary());
+            }
+        }
+    }
+
+    @Override
+    public void traverse() {
+        if(dictionaryHasMD5s(dictionaryToCompare)){
+            ReferenceSequenceTable table = new ReferenceSequenceTable(dictionaries);
+            table.build();
+
+            List<ReferencePair> refPairs = table.compareAgainstKeyReference(dictionaryPath);
+            for(ReferencePair pair : refPairs){
+                evaluateCompatibility(pair, table);
```

Comment:
this is a minor comment but i think its a little cleaner to have this method return an object that gets put into the map here rather than using a void method with side effects.